### PR TITLE
New route type metrics

### LINKED ISSUE #3: metrics: Add metrics for v1alpha1 Route types
- GRPCRoute
  - https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1alpha2/grpcroute_types.go
  - https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1alpha2.GRPCRoute
- TCPRoute
  - https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1alpha2/tcproute_types.go
  - https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1alpha2.TCPRoute
- TLSRoute
  - https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1alpha2/tlsroute_types.go
  - https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1alpha2.TLSRoute
- UDPRoute
  - https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1alpha2/udproute_types.go
  - https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1alpha2.UDPRoute

These reuse some of the same types as HTTProute, so the metrics can likely be reused and adapted.

### PR DESCRIPTION
**Pull request by Ygnas**
Closes: #3

### CODE REVIEW

File: config/gateway-api/kustomization.yaml [hunk 1]
```diff
@@ -2,4 +2,8 @@ resources:
 - crd/standard/gateway.networking.k8s.io_gatewayclasses.yaml
 - crd/standard/gateway.networking.k8s.io_gateways.yaml
 - crd/standard/gateway.networking.k8s.io_httproutes.yaml
+- crd/standard/gateway.networking.k8s.io_grpcroutes.yaml
+- crd/standard/gateway.networking.k8s.io_tcproutes.yaml
+- crd/standard/gateway.networking.k8s.io_tlsroutes.yaml
+- crd/standard/gateway.networking.k8s.io_udproutes.yaml
 # From https://github.com/kubernetes-sigs/gateway-api/blob/v0.6.2/config/crd/kustomization.yaml
```
Comment (david-martin) [inline config/gateway-api/kustomization.yaml:9]: Added https://github.com/Kuadrant/gateway-api-state-metrics/issues/26 while I noticed the older version 0.6.2 referenced

File: config/default/custom-resource-state.yaml [hunk 2]
```diff
@@ -188,6 +188,262 @@ spec:
...
+      - name: "hostname_info"
+        help: "Hostname information"
+        each:
+          type: Info
+          info:
+            path: [spec, hostnames]
+            labelsFromPath:
+              hostname: []
+      - name: "parent_info"
+        help: "Parent references that the httproute wants to be attached to"
```
Comment (david-martin) [inline config/default/custom-resource-state.yaml]: the help text here and in other resources looks like copy/paste from httproute

Comment (Ygnas) [reply config/default/custom-resource-state.yaml]: Fixed it

**Review by david-martin — APPROVED** [review]
Thanks @Ygnas 
Great to have these resources

**Pull request merged**